keanu-project/src/main/java/io/improbable/keanu/vertices/model/LambdaModelVertex.java

```diff
@@ -0,0 +1,114 @@
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class LambdaModelVertex extends DoubleVertex implements ModelVertex<DoubleTensor> {
+
+    private Map<VertexLabel, Vertex<? extends Tensor>> inputs;
+    private Map<VertexLabel, Tensor> outputs;
+    private Consumer<Map<VertexLabel, Vertex<? extends Tensor>>> executor;
+    private Function<Map<VertexLabel, Vertex<? extends Tensor>>, Map<VertexLabel, Tensor>> extractOutput;
+    private boolean hasValue;
+
+    /**
+     * A vertex whose operation is the execution of a lambda.
+     * It is able to execute a lambda and is able to parse the result.
+     * It stores multiple output values and a model result vertex is required to extract a specific value.
+     *
+     * @param inputs input vertices to the model
+     * @param executor the operation to perform
+     * @param updateValues a function to extract the output values (once the operation has been performed) and update
+     *                      the models output values.
+     */
+    public LambdaModelVertex(Map<VertexLabel, Vertex<? extends Tensor>> inputs,
+                             Consumer<Map<VertexLabel, Vertex<? extends Tensor>>> executor,
+                             Function<Map<VertexLabel, Vertex<? extends Tensor>>, Map<VertexLabel, Tensor>> updateValues) {
+        this.inputs = inputs;
+        this.outputs = Collections.emptyMap();
+        this.executor = executor;
+        this.extractOutput = updateValues;
+        this.hasValue = false;
+        setParents(inputs.values());
+    }
+
+    /**
+     * A vertex whose operation is the execution of a command line process.
+     * It is able to execute this process and parse the result.
+     * It stores multiple output values and a model result vertex is required to extract a specific value.
+     *
+     * @param inputs input vertices to the model
+     * @param command the command to execute
+     * @param updateValues a function to extract the output values (once the operation has been performed) and update
+     *                      the models output values.
+     * @return a process model vertex
+     */
+    public static LambdaModelVertex createFromProcess(Map<VertexLabel, Vertex<? extends Tensor>> inputs,
+                                           String command,
+                                           Function<Map<VertexLabel, Vertex<? extends Tensor>>, Map<VertexLabel, Tensor>> updateValues) {
+        return new LambdaModelVertex(inputs, i -> {
+            try {
+                Process cmd = Runtime.getRuntime().exec(command);
+                cmd.waitFor();
+            } catch (IOException | InterruptedException e) {
+                throw new RuntimeException("Failed during execution of the process. " + e);
+            }
+        }, updateValues);
+    }
+
+    @Override
+    public DoubleTensor calculate() {
+        run();
+        updateValues(inputs);
+        return DoubleTensor.scalar(0.0);
```

Comment:
why are you returning zero?

Reply:
Understood. I agree that we shouldn't enable vertices to have multiple values. But would it be OK to return the value of the first output vertex (ordered by VertexId)? Or - could the `LambdaModelVertex` receive the `VertexLabel` in its constructor so it knows how to return its output?

 If the answer is no - we need to document the semantics in javadoc here.